Elastica/EventSubscriber/UpdateContentIndexedSubscriber.php

```diff
@@ -33,14 +34,12 @@ public function updateIndexedContent(ContentEvent $event)
     {
         $content = $event->getContent();
 
-        if (!$content->getStatus()->isPublished()) {
-            return;
-        }
-
         $lastPublishedContent = $this->contentRepository->findLastPublishedVersion($content->getContentId(), $content->getLanguage());
 
-        if ($content->getVersion() >= $lastPublishedContent->getVersion()) {
-            $this->contentIndexor->index($content);
+        if ($lastPublishedContent instanceof ContentInterface) {
```

Comment:
If you unpublish a content, you will have to update the index to the last published content or you will have to purge the index from the content